infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/InfinispanRemoteDialect.java

```diff
@@ -0,0 +1,380 @@
+			}
+		}
+		else {
+			mapper.withinCacheEncodingContext( c -> c.put( idBuffer, valuePayload ) );
+		}
+	}
+
+	@Override
+	public void removeTuple(EntityKey key, TupleContext tupleContext) {
+		final String cacheName = key.getTable();
+		log.debugf( "removeTuple for key '%s' on cache '%s'", key, cacheName );
+		ProtoStreamMappingAdapter mapper = provider.getDataMapperForCache( cacheName );
+		ProtostreamId idBuffer = mapper.createIdPayload( key.getColumnNames(), key.getColumnValues() );
+		mapper.withinCacheEncodingContext( c -> c.remove( idBuffer ) );
+	}
+
+	@Override
+	public Association getAssociation(AssociationKey key, AssociationContext associationContext) {
+		Map<RowKey,Map<String, Object>> results = loadRowKeysByQuery( key );
+		return new Association( new MapAssociationSnapshot( results ) );
+	}
+
+	private Map<RowKey, Map<String, Object>> loadRowKeysByQuery(AssociationKey key) {
+		final String cacheName = key.getTable();
+		ProtostreamAssociationMappingAdapter mapper = provider.getCollectionsDataMapper( cacheName );
+		return mapper.withinCacheEncodingContext( c -> {
+			QueryFactory queryFactory = Search.getQueryFactory( c );
+			final String[] columnNames = key.getColumnNames();
+			QueryBuilder qb = queryFactory.from( ProtostreamPayload.class );
+			FilterConditionContext bqEnd = null;
+			boolean firstIteration = true;
+			for ( int i = 0; i < columnNames.length; i++ ) {
+				String fieldName = mapper.convertColumnNameToFieldName( columnNames[i] );
+				if ( firstIteration ) {
+					bqEnd = qb.having( fieldName ).eq( key.getColumnValues()[i] );
+					firstIteration = false;
+				}
+				else {
+					bqEnd = bqEnd.and().having( fieldName ).eq( key.getColumnValues()[i] );
+				}
+			}
+			Query query = bqEnd.toBuilder().build();
+			Map<RowKey,Map<String, Object>> resultsCollector = new HashMap<>();
+			try ( CloseableIterator<Entry<Object,Object>> iterator = c.retrieveEntriesByQuery( query, null, 100 ) ) {
+				while ( iterator.hasNext() ) {
+					Entry<Object,Object> e  = iterator.next();
+					ProtostreamPayload value = ( (ProtostreamPayload) e.getValue() );
+					Map<String, Object> entryObject = value.toMap();
+					RowKey entryKey = value.asRowKey( key );
+					resultsCollector.put( entryKey, entryObject );
+				}
+			}
+			return resultsCollector;
+		} );
+	}
+
+	@Override
+	public Association createAssociation(AssociationKey key, AssociationContext associationContext) {
+		Map<RowKey, Map<String, Object>> associationMap = new HashMap<RowKey, Map<String,Object>>();
+		return new Association( new MapAssociationSnapshot( associationMap ) );
+	}
+
+	@Override
+	public void insertOrUpdateAssociation(AssociationKey key, Association association, AssociationContext associationContext) {
+		final String cacheName = key.getTable();
+		final ProtoStreamMappingAdapter mapper = provider.getDataMapperForCache( cacheName );
+		final boolean embeddedInEntity = associatonTypeIsForeignKeyEmbeddedInEntity( key, association, mapper );
```

Comment:
What does that mean exactly? The many to one case? or rather `AssociationType.ONE_TO_ONE`